lib/promise.rb

```diff
@@ -19,7 +19,7 @@ def self.resolve(obj)
   end
 
   def self.all(enumerable)
-    Group.new(enumerable).promise
+    Group.new(new, enumerable).promise
```

Comment:
When I do that, then rubocop gives the following warning, so I think I will just keep it as is

```
lib/promise.rb:18:5: C: Redundant self detected.
    self.new.tap { |promise| promise.fulfill(obj) }
    ^^^^^^^^
```